iointel/src/agent_methods/tools_before_rebase/duckduckgo.py

```diff
@@ -0,0 +1,114 @@
+import asyncio
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, ConfigDict, model_validator
+
+
+class DuckDuckGoSearchAPIWrapper(BaseModel):
+    """Wrapper for DuckDuckGo Search API.
+
+    Free and does not require any setup.
+    """
+
+    region: Optional[str] = "wt-wt"
+    """
+    See https://pypi.org/project/duckduckgo-search/#regions
+    """
+    safesearch: str = "moderate"
+    """
+    Options: strict, moderate, off
+    """
+    time: Optional[str] = "y"
+    """
+    Options: d, w, m, y
+    """
+    max_results: int = 5
+    backend: str = "auto"
+    """
+    Options: auto, html, lite
+    """
+
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_environment(cls, values: Dict) -> Any:
+        """Validate that python package exists in environment."""
+        try:
+            from duckduckgo_search import DDGS  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "Could not import duckduckgo-search python package. "
+                "Please install it with `pip install -U duckduckgo-search`."
+            )
```

Comment:
```suggestion
        except ImportError as e:
            raise ImportError(
                "Could not import duckduckgo-search python package. "
                "Please install it with `pip install -U duckduckgo-search`."
            ) from e
```
let's retain original exception, too, to simplify diagnostics